pkg/kube/logger.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package kube provides Kubernetes client utilities for Helm.
+package kube
+
+import "log/slog"
+
```

Comment:
I am wondering where this should be, because I don't think it should be limited to `kube` package. Internal package?

Reply:
not sure what you mean?

Reply:
The issue I see is because interface for all helm packages use this slog adapter from `kube`, it will create unwanted links to kube package. I made a commit to show the issue (for me): https://github.com/helm/helm/commit/a1ca89b72a3ca45c81a6362dfd4bb6f122ade1ff#diff-a15c00615f24e3d585c1e7dee030d66e0e1d1666a34ce1ff038a84dbe2f5dba5L99

When I see [this](https://github.com/helm/helm/commit/a1ca89b72a3ca45c81a6362dfd4bb6f122ade1ff#diff-326ded87050c2e687ffc462ccc58f298102a4c3234645f356d1bd7350bda82d8R32), I don't think this is something we want.

I am wondering if
1. We want a slog adapter to still be able to pass a custom logger in packages ? 🙇🏻‍♂️
2. We want to have the slog adapter as an internal package like [`urlutil`](https://github.com/helm/helm/tree/main/internal/urlutil) ?

Reply:
Sorry for the late delay let me take a deeper look at it and we can bring this up at the next Helm developer meeting.